### ISSUE TITLE
Confusion with a list of union type

### ISSUE THREAD

**Issue by Leopard2A5** [aPureBase/KGraphQL#109]
```kotlin
import com.apurebase.kgraphql.KGraphQL
import kotlinx.coroutines.runBlocking
import org.junit.jupiter.api.Test

class InterfaceUnionTest {
    data class Wrapper(
        val items: List<QualificationItem>
    )

    sealed class QualificationItem {
        data class Qual1(
            val id: String
        ) : QualificationItem()
    }

    @Test
    fun test() {
        val schema = KGraphQL.schema {
            unionType<QualificationItem>()

            query("foo") {
                resolver { ->
                    Wrapper(listOf(QualificationItem.Qual1("12345")))
                }
            }
        }

        val foo = runBlocking {
            schema.execute("""
                query IntrospectionQuery {
                    __schema {
                        types {
                            kind
                            name
                        }
                    }
                }
            """)
        }
        println(foo)
    }
}
```

This fails with
```
An Object type must define one or more fields. Found none on type QualificationItem
com.apurebase.kgraphql.schema.SchemaException: An Object type must define one or more fields. Found none on type QualificationItem
```

The error disappears when you change the content of wrapper from a List<QualificationItem> to just QualificationItem.

Can't reproduce it, but i thought I'd seen QualificationItem show up as a type twice, once as an interface, once as a union.

**Comment by Leopard2A5** [comment]
If i change the definition of QualificationItem to
```kotlin
sealed class QualificationItem(val dummy: String = "dummy") {
    data class Qual1(
        val id: String
    ) : QualificationItem()
}
```

then there's no error, however the introspection query returns (some types omitted)
```json
{
    "data": {
        "__schema": {
            "types": [
                {
                    "kind": "INTERFACE",
                    "name": "QualificationItem"
                },
                {
                    "kind": "UNION",
                    "name": "QualificationItem"
                }
            ]
        }
    }
}
```

QualificationItem is defined once as an interface and once as a union.
I suspect that the interface definition comes from the type's usage in Wrapper and the union def comes from the explicit unionType call.

**Comment by Leopard2A5** [comment]
@jeggy what do you think?

**Comment by jeggy** [comment]
Thanks for these issues!
These are exactly the kind of testing we need more of 😃

**Comment by Leopard2A5** [comment]
Glad I could help, I really like the library 👍 
Keep up the good work!

**Comment by Leopard2A5** [comment]
Hope we can get a 0.15.3 soonish? This issue has been blocking me :)

**Comment by jeggy** [comment]
Yes, this will be available within 0.15.3 after [this pipeline](https://gitlab.com/apurebase/kgraphql/-/pipelines/172959237) finishes 👍
